src/app/parcels/form/ParcelForm.tsx

```diff
@@ -230,36 +224,23 @@ const ParcelForm: React.FC<ParcelFormProps> = ({
             last_updated: fields.lastUpdated,
         };
 
-        if (editMode) {
-            const { error: updateParcelError } = await updateParcel(formToAdd, parcelId!);
-            if (updateParcelError) {
-                let errorMessage: string;
-                switch (updateParcelError.type) {
-                    case "failedToUpdateParcel":
-                        errorMessage = `Failed to update parcel. Log ID: ${updateParcelError.logId}`;
-                        break;
-                    case "concurrentUpdateConflict":
-                        errorMessage = `Record has been edited recently - please refresh the page. LogID: ${updateParcelError.logId}`;
-                        break;
-                }
-                setSubmitErrorMessage(errorMessage);
-                return;
-            }
-        } else {
-            const { error: insertParcelError } = await insertParcel(formToAdd);
-            if (insertParcelError) {
-                let errorMessage: string;
-                switch (insertParcelError.type) {
-                    case "failedToInsertParcel":
-                        errorMessage = `Failed to insert parcel. Log ID: ${insertParcelError.logId}`;
-                        break;
-                }
-                setSubmitErrorMessage(errorMessage);
-                return;
+        const { parcelId, error } = await writeParcelInfoToDatabase(parcelRecord);
+
+        if (parcelId) {
+            router.push(parcelModalRouterPath(parcelId));
+        }
+
+        if (error) {
+            /*
+               If the user is trying to edit a parcel that is currently being edited,
+               we want the user to refresh the page to get the current data, so we keep the submit button
+               disabled.
+             */
```

Comment:
Yeah, I added as I thought it might be a little confusing as to why we're doing something different for one specific error but on second thoughts would be happy without as it's a bit long.